Libraries/Network/RCTDataManager.m

```diff
@@ -34,8 +35,37 @@ @implementation RCTDataManager
     NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
     request.HTTPMethod = [RCTConvert NSString:query[@"method"]] ?: @"GET";
     request.allHTTPHeaderFields = [RCTConvert NSDictionary:query[@"headers"]];
-    request.HTTPBody = [RCTConvert NSData:query[@"data"]];
 
+    NSString *dataURI = [RCTConvert NSString:query[@"dataURI"]];
+    NSData *data = [RCTConvert NSData:query[@"data"]];
+    if (dataURI != nil && data != nil) {
+      RCTLogError(@"You can only provide the 'data' OR 'dataURI' property on the query");
+      return;
+    }
+    if (data != nil) {
+      request.HTTPBody = data;
+      [RCTDataManager sendRequest:request responseSender:responseSender];
+      return;
+    }
+    if (dataURI != nil) {
+      if ([dataURI hasPrefix:@"assets-library"] ||
+          [dataURI hasPrefix:@"ph://"]) {
+        [RCTImageLoader loadImageWithTag:(NSString *)dataURI callback:^(NSError *error, UIImage *image) {
+          //TODO what to do with error?
```

Comment:
at least, RCTLogError it and don't execute the next two lines as they are going to crash the entire app?